src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -2301,31 +3027,73 @@ public enum RisingRivals implements LogicCardInfo {
           pokePower "Damage Switch", {
             text "As often as you like during your turn , you may move 1 damage counter from 1 of your Pokémon to another of your Pokémon . This power can’t be used if Alakazam 4 is affected by a Special Condition."
             actionA {
+              checkNoSPC()
+              assert my.all.find({it.numberOfDamageCounters>0}) : "None of your Pokémon have any damage counters"
+              assert my.all.size()>=2 : "You only have 1 Pokémon in play"
+              powerUsed()
+              def src=my.all.findAll {it.numberOfDamageCounters>0}.select("Source for damage counter")
+              def tar=my.all
+              tar.remove(src)
+              tar=tar.select("Target for damage counter")
+              src.damage-=hp(10)
+              directDamage 10, tar, SRC_ABILITY
+              bc "Swapped a damage counter from $src to $tar"
+              checkFaint()
             }
           }
           move "Mind Shock", {
             text "50 damage. This attack’s damage isn’t affected by Weakness or Resistance."
             energyCost P, P, C
             attackRequirement {}
             onAttack {
-              damage 0
+              noWrDamage 50
             }
           }
         };
       case FLOATZEL_GL_LV_X_104:
         return levelUp (this, from:"Floatzel GL", hp:HP100, type:WATER, retreatCost:0) {
           weakness L
           pokeBody "Water Rescue", {
-            text "Whenever any of your Pokémon (excluding any Floatzel) is Knocked Out by damage from your opponent’s attack, you may put that Pokémon and all cards that were attached to it from your discard pile into your hand."
-            delayedA {
+            text "Whenever any of your [W] Pokémon (excluding any Floatzel) is Knocked Out by damage from your opponent’s attack, you may put that Pokémon and all cards that were attached to it from your discard pile into your hand."
+            delayedA (priority: LAST) {
```

Comment:
Can you confirm this works correctly with a Fossil card underneath the W Pokémon? The scoopUpPokemon static does a bit of heavy lifting to handle ChangeImplementation cards like Fossils. You may need to use some of that logic here.